packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ArtifactController.java

```diff
@@ -135,17 +121,14 @@ public ResponseEntity<Artifact> getArtifact(@PathVariable("id") UUID artifactId)
 			@ApiResponse(responseCode = "500", description = "There was an issue updating the artifact", content = @Content)
 	})
 	public ResponseEntity<Artifact> updateArtifact(
-			@PathVariable("id") UUID artifactId,
-			@RequestBody Artifact artifact) {
+			@PathVariable("id") final UUID artifactId,
+			@RequestBody final Artifact artifact) {
 
 		try {
 			artifact.setId(artifactId);
-			Optional<Artifact> updated = artifactService.updateArtifact(artifact);
-			if (updated.isEmpty()) {
-				return ResponseEntity.notFound().build();
-			}
-			return ResponseEntity.ok(updated.get());
-		} catch (Exception e) {
+			final Optional<Artifact> updated = artifactService.updateAsset(artifact);
+			return updated.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.notFound().build());
```

Comment:
`notFound` returns a 404... you will need to update the ApiResponse

Reply:
fixed